integration_tests/steps/two_sided_pay_as_clear.py

```diff
@@ -105,20 +105,26 @@ def test_offer_bid_market_clearing_rate_files(context):
 
 @then('one-on-one matching of offer & bid in PAC happens at bid rate')
 def one_on_one_matching_at_clearing_rate_at_bid_rate(context):
-    assert all(isclose(clearing, 30.0)
-               for clearing in
-               bid_offer_matcher.matcher.match_algorithm.state.clearing.values())
+    count = 0
+    for clearing in bid_offer_matcher.matcher.match_algorithm.state.clearing.values():
```

Comment:
It looks like the content of `bid_offer_matcher.matcher.match_algorithm.state.clearing` is not documented anywhere in the code. Therefore it is quite complicated to understand what `cv[0]` is without getting deep into the chain of calls that generate it.
I think it would be good to add some docstrings both in these tests, and also here: https://github.com/gridsingularity/d3a-interface/blob/81469377497743b48cbae988f00862b0e19fccfb/d3a_interface/dataclasses.py#L87

But besides this: why aren't we using the `Clearing` dataclass inside the `clearing` dictionary? Here you use `cv[0]`, but it should actually be `cv.rate` (if we used the dataclass).

Reply:
> It looks like the content of bid_offer_matcher.matcher.match_algorithm.state.clearing is not documented anywhere in the code. Therefore it is quite complicated to understand what cv[0] is without getting deep into the chain of calls that generated it.

I guess as a software engineer, we can follow the chain to understand it.

> why aren't we using the Clearing dataclass inside the clearing dictionary?
clearing should be a Dict[market_id, Dict[DateTime, Clearing]], but right now it is not. In fact, here you use cv[0], but it should actually be cv.rate since we have a dataclass for it;

Have incorporated your suggestion.

> I think it would be good to add some docstrings both in these tests, and also here: https://github.com/gridsingularity/d3a-interface/blob/81469377497743b48cbae988f00862b0e19fccfb/d3a_interface/dataclasses.py#L87

Also incorporated